sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java

```diff
@@ -35,14 +35,19 @@ public void setBeanFactory(BeanFactory beanFactory) {
     }
 
     @Override
-    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @SuppressWarnings({ "rawtypes"})
     public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
-        if (bean instanceof AbstractAzureHttpClientBuilderFactory
-            && beanFactory.containsBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME)) {
+        if (!(bean instanceof AbstractAzureHttpClientBuilderFactory)) {
+            return bean;
+        }
+
+        if (beanFactory.containsBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME)) {
             HttpPipelinePolicy policy = (HttpPipelinePolicy) beanFactory.getBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME);
             AbstractAzureHttpClientBuilderFactory builderFactory = (AbstractAzureHttpClientBuilderFactory) bean;
             builderFactory.addHttpPipelinePolicy(policy);
             LOGGER.debug("Added the Sleuth http pipeline policy to {} builder.", bean.getClass());
+        } else {
+            LOGGER.warn("Not found the Sleuth http pipeline policy for {} builder.", bean.getClass());
```

Comment:
Actually, have you tried that if there's no sleuth pipeline bean in the beanFactory, how many times will this warning log be printed?